python/treelite/frontend.py

```diff
@@ -379,10 +379,15 @@ def from_xgboost(cls, booster):
                                 '`xgboost.Booster` object') from e
         if not isinstance(booster, xgboost.Booster):
             raise ValueError('booster must be of type `xgboost.Booster`')
-        if parse_version(xgboost.__version__) >= parse_version('1.0.0'):
-            # TODO(hcho3): Remove this line once XGBoost implements a method to save JSON model
-            # in-memory. (Right now, save_model() always saves to the disk.)
-            model_json_str = booster.__getstate__()['handle'].decode('utf-8')
+        xgb_version = parse_version(xgboost.__version__)
+        if xgb_version > parse_version("1.5.2"):
+            # For XGBoost version 1.6.0 and later, use save_raw() to export models as JSON string
+            model_json_str = booster.save_raw(raw_format="json").decode("utf-8")
```

Comment:
Is it necessary to decode it?

Reply:
@trivialfis No it is not necessary. I just need to revise `from_xgboost_json()` to accept a `bytearray` in addition to a string.